yt_dlp/extractor/epix.py

```diff
@@ -0,0 +1,169 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import uuid
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class EpixBase(InfoExtractor):
+    _API_URL = 'https://api.epix.com/v2/'
+    _API_KEY = 'f07debfcdf0f442bab197b517a5126ec'
+    _HEADERS = {
+        'user-agent': 'Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/65.0.3325.181 Chrome/65.0.3325.181 Safari/537.36',
+        'content-type': 'application/json'
+    }
+
+    def _real_initialize(self):
+        info = self._download_json(f'{self._API_URL}sessions', video_id='access token', data=json.dumps({
+            'device': {
+                'guid': uuid.uuid1().__str__(),
+                'format': 'console',
+                'os': 'web',
+                'app_version': '1.0.2',
+                'model': 'browser',
+                'manufacturer': 'google'
+            },
+            'apikey': self._API_KEY,
+            'oauth': {
+                'token': None
+            }
+        }).encode('utf-8'), headers=self._HEADERS, note='Download access token info')
+        access_token = traverse_obj(info, ('device_session', 'session_token'))
+        self._HEADERS.update({
+            'x-session-token': access_token
+        })
+
+
+class EpixExtraIE(EpixBase):
+    _VALID_URL = r'https?://www\.epix\.com/series/(?P<id>[^(?:/|\?)]+)(?:\/extra\/\d+|\?|$)'
+    IE_NAME = 'epix:extra'
+    IE_DESC = 'epix.com'
+    _TESTS = [{
+        'url': 'https://www.epix.com/series/from/extra/13596',
+        'md5': 'afedb9351282d7f8aee6e5474d8a7bc8',
+        'info_dict': {
+            'id': 'from',
+            'ext': 'mp4',
+            'title': 'FROM Critics Spot (Updated Trailer)',
+            'description': 'md5:a8ee8a70b0e4966c56675299a3d70951',
+            'thumbnail': r're:^https?://.+\.jpg',
+        },
+    }, {
+        'url': 'https://www.epix.com/series/billy-the-kid',
+        'md5': '100e5f60b16d39f1a5d9f1fe302d83ea',
+        'info_dict': {
+            'id': 'billy-the-kid',
+            'ext': 'mp4',
+            'title': 'Billy The Kid Trailer',
+            'description': 'md5:45b42f96458ee53b765e40d8d8c90998',
+            'thumbnail': r're:^https?://.+\.jpg',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        info = self._download_json(f'{self._API_URL}series/{video_id}', video_id=video_id, headers=self._HEADERS)
+        extras_content = traverse_obj(info, ('series', 'extras', 0, 'content'))
+
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(extras_content.get('hlspath'), video_id=video_id)
```

Comment:
`_extract_m3u8_formats...` will throw error with None. Also, the video cant be downloaded, and no data can be extracted without `extras_content`. So why use traverse_obj/.get here?

Similar issue below